File: reduce/reduce.js
// To run locally, use: node reduce.js

let arr = [{ prop: 1, name: 'first' }, { prop: 2, name: 'second' }, { prop: -2, name: 'third' }];

const res = arr.reduce((acc, obj) =>
  Math.abs(obj.prop) < Math.abs(acc.prop) ? obj : acc
);

console.log(res); // { prop: 1, name: 'first' }

File: reduce/reduce.rs
// To run locally, use: rustc reduce.rs && ./reduce

#[derive(Debug)]
struct Example {
    prop: i8,
    name: String,
}

fn main() {
    let arr = [
        Example { prop: 1, name: String::from("first") },
        Example { prop: 2, name: String::from("second") },
        Example { prop: -2, name: String::from("third") }
    ];

    let res = arr.iter().reduce(|acc, obj| {
      if (obj.prop).abs() < (acc.prop).abs() {
          obj
      } else {
          acc
      }
    });

    println!("{:?}", res.unwrap());  // Example { prop: 1, name: "first" }
}
